tests/gtest/aviflosslesstest.cc

```diff
@@ -115,80 +161,11 @@ TEST_P(LosslessTest, EncodeDecode) {
-                    /*allow_changing_cicp=*/false, /*ignore_gain_map=*/false,
-                    AVIF_DEFAULT_IMAGE_SIZE_LIMIT, image.get(),
-                    /*outDepth=*/nullptr, /*sourceTiming=*/nullptr,
-                    /*frameIter=*/nullptr) == AVIF_APP_FILE_FORMAT_UNKNOWN) {
-    return nullptr;
-  }
-  return image;
-}
-
-// Tests encode/decode round trips under different matrix coefficients.
-TEST_P(LosslessTest, EncodeDecodeMatrixCoefficients) {
-  const std::string& file_name = std::get<0>(GetParam());
-  const avifMatrixCoefficients matrix_coefficients =
-      static_cast<avifMatrixCoefficients>(std::get<1>(GetParam()));
-  const avifPixelFormat pixel_format =
-      static_cast<avifPixelFormat>(std::get<2>(GetParam()));
-  // Ignore ICC when going from RGB to gray.
-  const avifBool ignore_icc =
-      pixel_format == AVIF_PIXEL_FORMAT_YUV400 ? AVIF_TRUE : AVIF_FALSE;
-
-  // Read a ground truth image but ask for certain matrix coefficients.
-  ImagePtr image;
-  ASSERT_NO_FATAL_FAILURE(ReadImageSimple(
-      file_name, pixel_format, matrix_coefficients, ignore_icc, image));
-  if (image == nullptr) return;
-
-  // Convert to a temporary PNG and read back losslessly.
-  const std::string tmp_path =
-      testing::TempDir() + "decoded_EncodeDecodeMatrixCoefficients.png";
-  ASSERT_TRUE(testutil::WriteImage(image.get(), tmp_path.c_str()));
-  const ImagePtr decoded_lossless =
-      ReadImageLossless(tmp_path, pixel_format, ignore_icc);
-  if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV420) {
-    // 420 cannot be converted from RGB to YUV with
-    // AVIF_MATRIX_COEFFICIENTS_IDENTITY due to a decision taken in
-    // avifGetYUVColorSpaceInfo.
-    ASSERT_EQ(decoded_lossless, nullptr);
-    return;
-  }
-  ASSERT_NE(decoded_lossless, nullptr);
-
-  // Verify that the ground truth and decoded images are the same.
-  const ImagePtr ground_truth_lossless = ReadImageLossless(
-      std::string(data_path) + file_name, pixel_format, ignore_icc);
-  ASSERT_NE(ground_truth_lossless, nullptr);
-  const bool are_images_equal =
-      testutil::AreImagesEqual(*ground_truth_lossless, *decoded_lossless);
-
-  if (matrix_coefficients == AVIF_MATRIX_COEFFICIENTS_YCGCO) {
-    // AVIF_MATRIX_COEFFICIENTS_YCGCO is not a lossless transform.
-    ASSERT_FALSE(are_images_equal);
-  } else if (matrix_coefficients == AVIF_MATRIX_COEFFICIENTS_YCGCO_RE &&
-             pixel_format == AVIF_PIXEL_FORMAT_YUV400) {
-    // For gray images, information is lost in YCGCO_RE.
-    ASSERT_FALSE(are_images_equal);
-  } else {
-    ASSERT_TRUE(are_images_equal);
-  }
-}
-
 INSTANTIATE_TEST_SUITE_P(
-    LosslessTestInstantiation, LosslessTest,
+    EncodeDecodeRamInstantiation, EncodeDecodeRam,
     testing::Combine(
-        testing::Values("paris_icc_exif_xmp.png", "paris_exif_xmp_icc.jpg"),
+        testing::Values("paris_icc_exif_xmp.png", "paris_exif_xmp_icc.jpg",
+                        "kodim03_grayscale_gamma1.6.png"),
```

Comment:
any source file with GRAYA coverage?

Reply:
Noe that I could find but the roundtrip is tested in avifreadimagetest.cc